src/rgw/rgw_data_sync.cc

```diff
@@ -2776,8 +2776,16 @@ class RGWInitBucketShardSyncStatusCoroutine : public RGWCoroutine {
             status.state = rgw_bucket_shard_sync_info::StateFullSync;
             status.inc_marker.position = info.max_marker;
           } else {
+            const bool reinit = status.state == rgw_bucket_shard_sync_info::StateReInit;
+            if (reinit) {
+              // continue processing at the end, though we may skip entries
+              // after the CLS_RGW_OP_RESYNC
+              status.inc_marker.position = info.max_marker;
```

Comment:
ideally we wouldn't have to set this at the end, we could just leave it at the SYNCSTOP position. then we'd resume from the RESYNC op and sync everything after. need to test whether that works reliably